megatron/core/transformer/transformer_config.py

```diff
@@ -164,6 +164,9 @@ class TransformerConfig(ModelParallelConfig):
     apply_rope_fusion: bool = False
     """If True, use fused RoPE kernel."""
 
+    use_te_fused_rope: bool = False
```

Comment:
@gurpreet-dhami @lcskrishna I would suggest using TE fused RoPE by default. Most users don't care about which RoPE implementation they are using, but they want performance. And let's give them the performant kernel by default. And in case TE is not installed in the docker, we have a mechanism to fall back to the unfused RoPE.

Reply:
Yes we can enable it by default. @gurpreet-dhami Can you make the change.